src/azure-cli/azure/cli/command_modules/appconfig/_featuremodels.py

```diff
@@ -129,86 +167,502 @@ def __repr__(self):
+                )
+                allocation_percentile_list.append(feature_percentile_allocation)
+
+            allocation.percentile = allocation_percentile_list
+
+        return allocation
+
+    def __repr__(self):
+        featureallocation = {
+            FeatureFlagConstants.USER: list(self.user),
+            FeatureFlagConstants.GROUP: list(self.group),
+            FeatureFlagConstants.PERCENTILE: list(self.percentile),
+            FeatureFlagConstants.DEFAULT_WHEN_ENABLED: self.default_when_enabled,
+            FeatureFlagConstants.DEFAULT_WHEN_DISABLED: self.default_when_disabled,
+            FeatureFlagConstants.SEED: self.seed
+        }
+
+        return json.dumps(featureallocation, indent=2, ensure_ascii=False)
+
+
+# Feature Flag Helper Functions #
+def custom_serialize_conditions(conditions_dict):
+    """
+    Helper Function to serialize Conditions
+
+    Args:
+        conditions_dict - Dictionary of {str, Any}
+
+    Return:
+        JSON serializable Dictionary
+    """
     featurefilterdict = {}
 
     for key, value in conditions_dict.items():
         if key == FeatureFlagConstants.CLIENT_FILTERS:
-            featurefilterdict[key] = [feature_filter.__dict__ for feature_filter in value]
+            featurefilterdict[key] = [
+                feature_filter.__dict__ for feature_filter in value
+            ]
         else:
             featurefilterdict[key] = value
 
     return featurefilterdict
 
 
+def custom_serialize_allocation(allocation):
+    """
+    Helper Function to serialize Allocation
+
+    Args:
+        allocation_dict - FeatureAllocation object
+
+    Return:
+        JSON serializable Dictionary
+    """
+    featureallocationdict = {}
+
+    if allocation.user:
+        featureallocationdict[FeatureFlagConstants.USER] = [
+            user.__dict__ for user in allocation.user
+        ]
+
+    if allocation.group:
+        featureallocationdict[FeatureFlagConstants.GROUP] = [
+            group.__dict__ for group in allocation.group
+        ]
+
+    if allocation.percentile:
+        featureallocationdict[FeatureFlagConstants.PERCENTILE] = [
+            percentile.__dict__ for percentile in allocation.percentile
+        ]
+
+    if allocation.default_when_enabled:
+        featureallocationdict[FeatureFlagConstants.DEFAULT_WHEN_ENABLED] = allocation.default_when_enabled
+
+    if allocation.default_when_disabled:
+        featureallocationdict[FeatureFlagConstants.DEFAULT_WHEN_DISABLED] = allocation.default_when_disabled
+
+    return featureallocationdict
+
+
+def custom_serialize_variants(variants_list):
+    """
+    Helper Function to serialize Variants
+
+    Args:
+        variants_list - List of FeatureVariant objects
+
+    Return:
+        JSON serializable List
+    """
+    featurevariants = []
+    for variant in variants_list:
+        variant_dict = {}
+        variant_dict[FeatureFlagConstants.NAME] = variant.name
+        if variant.configuration_value:
```

Comment:
configuration_value can be false, empty string or null, and all of these are valid values that should be added to the feature flag. This `if` condition will block such values.

Reply:
You are right. Updated